Cannot read bookmark list

### Describe your problem!

A single user has a problem, that he (PM) can not access the bookmark lists but gets an error toast, that it can't be read.

I asked for a debug log.
With activated debug logging the app crashes when trying to open the List/PQ menu.

Stack trace:
```
9-04 13:00:29.091 18001 18060 E AndroidRuntime: FATAL EXCEPTION: network--5
09-04 13:00:29.091 18001 18060 E AndroidRuntime: Process: cgeo.geocaching, PID: 18001
09-04 13:00:29.091 18001 18060 E AndroidRuntime: io.reactivex.rxjava3.exceptions.UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | java.lang.RuntimeException: Aborting on Log.e()
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at io.reactivex.rxjava3.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:372)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:325)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler$ExecutorWorker.runEager(ExecutorScheduler.java:287)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:248)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:929)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: Caused by: java.lang.RuntimeException: Aborting on Log.e()
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.utils.Log.e(Log.java:325)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.GCParser.searchBookmarkLists(GCParser.java:1069)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.PocketQueryListActivity.lambda$loadInBackground$1(PocketQueryListActivity.java:83)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.PocketQueryListActivity.lambda$loadInBackground$1$PocketQueryListActivity(Unknown Source:0)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.-$$Lambda$PocketQueryListActivity$wkb8AucIHdJVHtkSLkrnzscYwIs.run(Unknown Source:2)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.utils.AndroidRxUtils.lambda$andThenOnUi$2(AndroidRxUtils.java:70)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.utils.-$$Lambda$AndroidRxUtils$bqzWo9nlvU86BvboB-9jhDWp5vc.run(Unknown Source:4)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:322)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	... 5 more
09-04 13:00:29.091 18001 18060 E AndroidRuntime: Caused by: java.text.ParseException: Unparseable date: "2021-08-22T21:12:35Z"
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at java.text.DateFormat.parse(DateFormat.java:362)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.utils.SynchronizedDateFormat.parse(SynchronizedDateFormat.java:22)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.GCParser.searchBookmarkLists(GCParser.java:1061)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	... 11 more
09-04 13:01:17.548 18489 18629 E AndroidRuntime: FATAL EXCEPTION: network--7
09-04 13:01:17.548 18489 18629 E AndroidRuntime: Process: cgeo.geocaching, PID: 18489
09-04 13:01:17.548 18489 18629 E AndroidRuntime: io.reactivex.rxjava3.exceptions.UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | java.lang.RuntimeException: Aborting on Log.e()
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at io.reactivex.rxjava3.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:372)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:325)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler$ExecutorWorker.runEager(ExecutorScheduler.java:287)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:248)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:929)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: Caused by: java.lang.RuntimeException: Aborting on Log.e()
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at cgeo.geocaching.utils.Log.e(Log.java:325)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.GCParser.searchBookmarkLists(GCParser.java:1069)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.PocketQueryListActivity.lambda$loadInBackground$1(PocketQueryListActivity.java:83)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.PocketQueryListActivity.lambda$loadInBackground$1$PocketQueryListActivity(Unknown Source:0)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.-$$Lambda$PocketQueryListActivity$wkb8AucIHdJVHtkSLkrnzscYwIs.run(Unknown Source:2)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at cgeo.geocaching.utils.AndroidRxUtils.lambda$andThenOnUi$2(AndroidRxUtils.java:70)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at cgeo.geocaching.utils.-$$Lambda$AndroidRxUtils$bqzWo9nlvU86BvboB-9jhDWp5vc.run(Unknown Source:4)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:322)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	... 5 more
09-04 13:01:17.548 18489 18629 E AndroidRuntime: Caused by: java.text.ParseException: Unparseable date: "2021-08-22T21:12:35Z"
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at java.text.DateFormat.parse(DateFormat.java:362)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at cgeo.geocaching.utils.SynchronizedDateFormat.parse(SynchronizedDateFormat.java:22)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.GCParser.searchBookmarkLists(GCParser.java:1061)
09-04 13:01:17.548 18489 18629 E AndroidRuntime: 	... 11 more
```

### How to reproduce?

Main menu - Open Lists/Pocket-Queries overview

### Actual result after these steps?

Error toast "cannot read bookmark list"

### Expected result after these steps?

Bookmark lists shown

### Reproducible

No

### c:geo Version

202108.28

### System information

```text
## System information c:geo version: 2021.08.28 Device: ------- - Device type: SNE-LX1 (SNE-LX1, HUAWEI) - Available processors: 8 - Android version: 10 - Android build: SNE-LX1 10.0.0.245(C432E10R1P1) - Screen resolution: 1080x2128px (360x709dp) - Sailfish OS detected: false - Google Play services: enabled - 21.30.16 (120408-391784508) - HW acceleration: enabled (default state) Sensor and location: ------- - Low power mode: inactive - Compass capabilities: yes - Rotation vector sensor: present - Orientation sensor: present - Magnetometer & Accelerometer sensor: present - Direction sensor used: orientation Program settings: ------- - Settings: v8, Count:150 - Set language: de - System date format: dd.MM.yy - Time zone: GMT+02:00 - Debug mode active: yes - Last backup: never - Routing mode: Walk - Live map mode: true - OSM multi-threading: true / threads: 4 - Map: Google: Satellit - Id: cgeo.geocaching.maps.google.v2.GoogleMapProvider$GoogleSatelliteSource - Atts: none - Theme: none Filters: ------- - Hide waypoints: - - LIVE: Status:4 Elemente ([:inconclusive=false:advanced=false]AND(type;difficulty_terrain:d=-:d=-:t=-:t=-;status:owned_no:found_no:exclude_disabled:exclude_archived)) - OFFLINE: Keine ([:inconclusive=false:advanced=false]AND(type;difficulty_terrain:d=-:d=-:t=-:t=-;status)) Services: ------- - Geocaching sites enabled: geocaching.com: Logged in (Anmeldung OK) / PREMIUM Geocaching.com Adventure Lab opencaching.de: Not logged in (Nicht autorisiert) - Geocaching.com date format: dd/MM/yyyy - Routing: external / BRouter installed: false - Installed c:geo plugins: none Permissions & paths: ------- - Fine location permission: granted - Write external storage permission: granted - System internal c:geo dir: /data/user/0/cgeo.geocaching (21,8 GB free) v2 internal isDir(8 entries) - Legacy User storage c:geo dir: /storage/emulated/0/cgeo (21,8 GB free) v2 external non-removable isDir(2 entries) - Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (21,8 GB free) v2 external non-removable isDir(1547 entries) - Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (21,8 GB free) v2 internal isDir(0 entries) - Public Folders: #11 - BASE: SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021 (User-Defined)[SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021::]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021/document/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021, Av:true, files:>=191, dirs:>=9, totalFileSize:>=61,5 MB, free space: 13,8 GB, files on device: 0) - OFFLINE_MAPS: SanDisk-Speicherkarte/C Geo Maps offline (User-Defined)[SanDisk-Speicherkarte/C Geo Maps offline [DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/3633-3431%3AC%20Geo%20Maps%20offline%20::]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3AC%20Geo%20Maps%20offline%20/document/3633-3431%3AC%20Geo%20Maps%20offline%20, Av:true, files:12, dirs:1, totalFileSize:0 B, free space: 13,8 GB, files on device: 0) - OFFLINE_MAP_THEMES: SanDisk-Speicherkarte/C Geo Maps offline /_themes (Default)[SanDisk-Speicherkarte/C Geo Maps offline /_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/3633-3431%3AC%20Geo%20Maps%20offline%20::/_themes]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3AC%20Geo%20Maps%20offline%20/document/3633-3431%3AC%20Geo%20Maps%20offline%20%2F_themes, Av:true, files:12, dirs:0, totalFileSize:0 B, free space: 13,8 GB, files on device: 0) - LOGFILES: SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/logfiles (Default)[SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021/document/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2Flogfiles, Av:true, files:16, dirs:0, totalFileSize:109,5 KB, free space: 13,8 GB, files on device: 0) - GPX: SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/gpx (User-Defined)[SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/gpx[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2Fgpx::]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2Fgpx/document/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2Fgpx, Av:true, files:151, dirs:0, totalFileSize:61,4 MB, free space: 13,8 GB, files on device: 0) - BACKUP: SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/backup (Default)[SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021::/backup]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021/document/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2Fbackup, Av:true, files:12, dirs:0, totalFileSize:0 B, free space: 13,8 GB, files on device: 0) - FIELD_NOTES: SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/field-notes (Default)[SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021/document/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2Ffield-notes, Av:true, files:12, dirs:0, totalFileSize:0 B, free space: 13,8 GB, files on device: 0) - SPOILER_IMAGES: SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/GeocachePhotos (Default)[SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021/document/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2FGeocachePhotos, Av:true, files:12, dirs:3, totalFileSize:0 B, free space: 13,8 GB, files on device: 0) - ROUTING_BASE: SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/routing (Default)[SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021::/routing]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021/document/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2Frouting, Av:true, files:19, dirs:1, totalFileSize:81,2 KB, free space: 13,8 GB, files on device: 0) - ROUTING_TILES: SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/routing/segments4 (Default)[SanDisk-Speicherkarte/cgeo/C.Geo neu April 2021/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021/document/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 13,8 GB, files on device: 0) - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 21,8 GB, files on device: -1) - Map render theme path: - PersistedDocumentUris: #1 - TRACK: null - Persisted Uri Permissions: #3 - content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021%2Fgpx (15. Juli, 18:50):RW - content://com.android.externalstorage.documents/tree/3633-3431%3Acgeo%2FC.Geo%20neu%20April%202021 (22. Apr., 18:39):RW - content://com.android.externalstorage.documents/tree/3633-3431%3AC%20Geo%20Maps%20offline%20 (22. Apr., 18:41):RW - Database: /data/user/0/cgeo.geocaching/databases/data (v96, Size:60,3 MB) on system internal storage --- End of system information ---
```


### Additional Information

I cannot reproduce and it also works for the users partner (on another account) with same device.
Seems to be account specific. Date format?

Interesting part:

```
Caused by: java.text.ParseException: Unparseable date: "2021-08-22T21:12:35Z"
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at java.text.DateFormat.parse(DateFormat.java:362)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.utils.SynchronizedDateFormat.parse(SynchronizedDateFormat.java:22)
09-04 13:00:29.091 18001 18060 E AndroidRuntime: 	at cgeo.geocaching.connector.gc.GCParser.searchBookmarkLists(GCParser.java:1061)

--- a/main/src/cgeo/geocaching/connector/gc/GCParser.java
+++ b/main/src/cgeo/geocaching/connector/gc/GCParser.java
@@ -90,6 +90,7 @@ public final class GCParser {
 
     @NonNull
     private static final SynchronizedDateFormat DATE_JSON = new SynchronizedDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", TimeZone.getTimeZone("UTC"), Locale.US); // 2009-03-28T18:30:31.497Z
+    private static final SynchronizedDateFormat DATE_JSON_SHORT = new SynchronizedDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", TimeZone.getTimeZone("UTC"), Locale.US); // 2009-03-28T18:30:31Z
 
     @NonNull
     private static final ImmutablePair<StatusCode, Geocache> UNKNOWN_PARSE_ERROR = ImmutablePair.of(StatusCode.UNKNOWN_ERROR, null);
@@ -1016,7 +1017,15 @@ public final class GCParser {
                 final String name = row.get("name").asText();
                 final String guid = row.get("referenceCode").asText();
                 final int count = row.get("count").asInt();
-                final Date date = DATE_JSON.parse(row.get("lastUpdateUtc").asText());
+                Date date;
+                final String lastUpdateUtc = row.get("lastUpdateUtc").asText();
+                try {
+                    date = DATE_JSON.parse(lastUpdateUtc);
+                } catch (ParseException e) {
+                    // if parsing with fractions of seconds failed, try short form
+                    date = DATE_JSON_SHORT.parse(lastUpdateUtc);
+                    Log.d("parsing bookmark list: fallback needed for '" + lastUpdateUtc + "'");
+                }
 
                 final PocketQuery pocketQuery = new PocketQuery(guid, name, count, true, date.getTime(), -1, true);
                 list.add(pocketQuery);

